wagtail/contrib/settings/tests/site_specific/test_model.py

```diff
@@ -18,6 +18,19 @@ def test_for_site_returns_expected_settings(self):
             with self.subTest(site=site):
                 self.assertEqual(TestSiteSetting.for_site(site), expected_settings)
 
+    @override_settings(ALLOWED_HOSTS=["no-site-match.example"])
+    def test_for_request_raises_does_not_exist_when_no_site_match(self):
+        Site.objects.update(is_default_site=False)
+        # Use RequestFactory directly, as self.get_request sets SERVER_NAME and site.
+        request = RequestFactory(SERVER_NAME="no-site-match.example").get("/")
+        with self.assertRaises(TestSiteSetting.DoesNotExist):
+            TestSiteSetting.for_request(request)
+
+    @override_settings(ALLOWED_HOSTS=["no-site-match.example"])
```

Comment:
**Suggestion**: I think this decorator is copypasta.

```suggestion
```

Reply:
I agree... removed in 6b38cbfec9.